ddsketch/mapping/cubically_interpolated_mapping.go

```diff
@@ -0,0 +1,131 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2020 Datadog, Inc.
+
+package mapping
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"math"
+
+	"github.com/DataDog/sketches-go/ddsketch/pb/sketchpb"
+)
+
+const (
+	// Assuming we write the index as index(v) = floor(multiplier*ln(2)/ln(gamma)*(e+As^3+Bs^2+Cs)), where v=2^e(1+s)
+	// and gamma = (1+relativeAccuracy)/(1-relativeAccuracy), those are the coefficients that minimize the multiplier,
+	// therefore the memory footprint of the sketch, while ensuring the relative accuracy of the sketch.
+	A = 6.0 / 35.0
+	B = -3.0 / 5.0
+	C = 10.0 / 7.0
+)
+
+type CubicallyInterpolatedMapping struct {
+	relativeAccuracy      float64
+	multiplier            float64
+	normalizedIndexOffset float64
+}
+
+func NewCubicallyInterpolatedMapping(relativeAccuracy float64) (*CubicallyInterpolatedMapping, error) {
+	if relativeAccuracy <= 0 || relativeAccuracy >= 1 {
+		return nil, errors.New("The relative accuracy must be between 0 and 1.")
+	}
+	return &CubicallyInterpolatedMapping{
+		relativeAccuracy: relativeAccuracy,
+		multiplier:       7.0 / (10 * math.Log((1+relativeAccuracy)/(1-relativeAccuracy))),
```

Comment:
Alternatively (this is more accurate):
```suggestion
		multiplier:       7.0 / (10 * math.Log1p(2*relativeAccuracy/(1-relativeAccuracy))),
```